pkg/controller/seed-controller-manager/cni-application-installation-controller/controller.go

```diff
@@ -369,6 +369,12 @@ func ApplicationInstallationReconciler(cluster *kubermaticv1.Cluster, overwriteR
 				return app, fmt.Errorf("failed to merge CNI values: %w", err)
 			}
 
+			if cluster.Spec.CNIPlugin.Type == kubermaticv1.CNIPluginTypeCilium {
+				ipam := values["ipam"].(map[string]any)
+				operator := ipam["operator"].(map[string]any)
+				delete(operator, "clusterPoolIPv4PodCIDR")
+			}
```

Comment:
Will this work? Can you maybe test against the upgrade environment? If the field is removed, I suspect it might be omitted from the patch and won't change?

Reply:
It might be the case here, let me check it first then